hooks/pre-command

```diff
@@ -42,40 +47,69 @@ upsert_ecr() {
   echo "File: ${lifecycle_policy_file}"
   aws ecr put-lifecycle-policy \
   --repository-name "${repository_name}" \
+  --region "$region" \
   --lifecycle-policy-text "file://${lifecycle_policy_file}"
 
   repository_tags_file="${BUILDKITE_PLUGIN_CREATE_ECR_REPOSITORY_TAGS_FILE:-}"
   if [[ ${repository_tags_file} != '' ]]; then
     echo '--- Setting ECR repository tags'
     echo "File: ${repository_tags_file}"
     echo '--- Getting ECR repository Arn'
-    get_repository_arn "${repository_name}"
+    get_repository_arn "${repository_name}" "${region}"
     echo "Repository Arn: ${repository_arn}"
-    aws ecr tag-resource \
-      --resource-arn "${repository_arn}" \
-      --tags "file://${repository_tags_file}"
+    aws ecr tag-resource --resource-arn "${repository_arn}" --tags "file://${repository_tags_file}" --region "${region}" 
   fi
 
   if [[ "${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH:-}" =~ ^(true|on|1)$ ]]; then
     echo '--- Setting ECR image scanning configuration as enabled'
     aws ecr put-image-scanning-configuration \
       --repository-name "${repository_name}" \
+      --region "$region" \
       --image-scanning-configuration \
       scanOnPush=true
   elif [[ "${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH:-}" =~ ^(false|off|0)$ ]]; then
     echo '--- Setting ECR image scanning configuration as disabled'
     aws ecr put-image-scanning-configuration \
       --repository-name "${repository_name}" \
+      --region "$region" \
       --image-scanning-configuration \
       scanOnPush=false
   fi
 }
 
+upsert_to_regional_ecr() {
+  local repository_name="${1}"
+  shift
+  local regions=("$@")
+  # run a loop for each region mentioned in the list
+  for region in "${regions[@]}"; do
+    echo "--- Current region => ${region}"
+    upsert_ecr "${repository_name}" "${region}"
+  done
+}
+
+get_regions() {
+  regions=()
+
+  if [ -z "${BUILDKITE_PLUGIN_CREATE_ECR_REGIONS_0:-}" ]; then
+    echo "--- No regions mentioned. Getting current region"
+    fallback_region=$(aws ec2 describe-availability-zones --output text --query 'AvailabilityZones[0].[RegionName]')
```

Comment:
Is it possible for a build agent to be running unprivileged without `ec2:DescribeAvailabilityZones`? I wonder if something like `"${AWS_DEFAULT_REGION:-$(aws configure get region)}"` would suffice as a backward-compatible alternative.

Reply:
Yeah, I guess this would be better. Thanks for the suggestion. Sending in an update.